lib/src/client/http2_connection.dart

```diff
@@ -81,7 +81,7 @@ class Http2ClientConnection implements connection.ClientConnection {
     if (securityContext != null) {
       // Todo(sigurdm): We want to pass supportedProtocols: ['h2']. http://dartbug.com/37950
       socket = await SecureSocket.secure(socket,
-          host: authority,
+          host: options.credentials.authority ?? host,
```

Comment:
@sigurdm Since there was all this confusion around the HTTP2 authority and "this" authority (which as far as I can tell is always without a port and just a way to allow certs from a different host), should there maybe be a comment on `authority` to disambiguate it from the other `authority` field?


Also is maybe the `host` parameter name on `SecureSocket.secure` misleading and should rather the `authority`? Because the actual connection to the `host` already gets made in line 80 (`Socket.connect(host, port);`).

Reply:
Good idea to leave a comment.

I don't think we can change the argument name on `SecureSocket` easily.

Reply:
It might be nice with a comment here explaining that the `host` is actually the `authority` we're verifying the TLS connection to belong to.

Reply:
Done